markdown/postprocessors.py

```diff
@@ -73,37 +72,26 @@ class RawHtmlPostprocessor(Postprocessor):
 
     def run(self, text: str) -> str:
         """ Iterate over html stash and restore html. """
-        replacements = OrderedDict()
-        for i in range(self.md.htmlStash.html_counter):
-            html = self.stash_to_string(self.md.htmlStash.rawHtmlBlocks[i])
-            if self.isblocklevel(html):
-                replacements["<p>{}</p>".format(
-                    self.md.htmlStash.get_placeholder(i))] = html
-            replacements[self.md.htmlStash.get_placeholder(i)] = html
-
         def substitute_match(m: re.Match[str]) -> str:
-            key = m.group(0)
-
-            if key not in replacements:
-                if key[3:-4] in replacements:
-                    return f'<p>{ replacements[key[3:-4]] }</p>'
-                else:
-                    return key
-
-            return replacements[key]
-
-        if replacements:
+            if key := m.group(1):
+                wrapped = True
+            else:
+                key = m.group(2)
+                wrapped = False
+            if (key := int(key)) >= len(self.md.htmlStash.rawHtmlBlocks):
```

Comment:
Could use `html_counter` instead:

```suggestion
            if (key := int(key)) >= self.md.htmlStash.html_counter:
```

Reply:
I don't see any reason to recompute the length when we already have it stored.

What's not clear to me is why we are reassigning `key` here. I understand the need to convert a string to an integer, but why do that here? Why not when it is initially assigned on lines 76 and/or 79?

Reply:
Oh, just an oversight, we can definitely coerce to int when first assigning.

Reply:
Ah I remember now, that's because I'm doing `if key := m.group(1)`, and casting to `int` here would fail when the `m.group(1)` is `None`.

Reply:
Let me know if you have another idea to make this more elegant, I'll mark as resolved for now :slightly_smiling_face: